### PR TITLE
[Enhancement] Improve stability in parquet reader footer parse 

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Problem Summary(Required) ：

In this pr:
* More accurately stat count
* More struct bound condition check
* If face encrypted parquet file, throw not support error
* Avoid be crash when read 0 byte parquet file

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/formats/parquet/file_reader.cpp [hunk 1]
```diff
@@ -67,60 +67,69 @@ Status FileReader::init(HdfsScannerContext* ctx) {
...
-    uint8_t* footer_buf = local_buf;
-    // we may allocate on heap if local_buf is not large enough.
-    DeferOp deferop([&] {
-        if (footer_buf != local_buf) {
-            delete[] footer_buf;
-        }
-    });
+    std::vector<char> footer_buffer;
+    ASSIGN_OR_RETURN(uint32_t footer_read_size, _get_footer_read_size());
+    footer_buffer.resize(footer_read_size);
```
Comment (imay) [inline be/src/formats/parquet/file_reader.cpp:72]: I think you can use struct in `be/src/util/raw_container.h` to ignore unnecessary memset(0)

File: be/src/formats/parquet/file_reader.cpp [hunk 2]
```diff
@@ -67,60 +67,69 @@ Status FileReader::init(HdfsScannerContext* ctx) {
...
+StatusOr<uint32_t> FileReader::_get_footer_read_size() const {
+    if (_file_size == 0) {
+        return Status::Corruption("Parquet file size is 0 bytes");
+    } else if (_file_size < PARQUET_FOOTER_SIZE) {
+        return Status::Corruption(strings::Substitute(
+                "Parquet file size is $0 bytes, smaller than the minimum parquet file footer ($1 bytes)", _file_size,
+                PARQUET_FOOTER_SIZE));
     }
-    return Status::OK();
+    return std::min(_file_size, DEFAULT_FOOTER_BUFFER_SIZE);
```
Comment (letian-jiang) [inline be/src/formats/parquet/file_reader.cpp:113]: what if file_size is greater than DEFAULT_FOOTER_BUFFER_SIZE?

**Review by letian-jiang — APPROVED** [review]
LGTM

**Review by DorianZheng — APPROVED** [review]

**Comment by DorianZheng** [comment]
Please add problem summary section

**Comment by letian-jiang** [comment]
I'd like to know why this PR could improve the stability of parsing parquet footer.

File: be/src/formats/parquet/file_reader.cpp (hunk 1 above)
Comment (Smith-Cruise) [reply be/src/formats/parquet/file_reader.cpp:72]: Improve in next pr.

**Review by dirtysalt — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**